### PR TITLE
[116] Enhance display of Table to use Eclipse General preferences

### PR DESCRIPTION
**Pull request by mypsycho**
Ticket #116

### CODE REVIEW

File: plugins/org.eclipse.sirius.tests.sample.xtext.statemachine/.gitignore
Comment (lredor) [inline plugins/org.eclipse.sirius.tests.sample.xtext.statemachine/.gitignore:1]: We are OK that this commit has no link with the current PR and issue? In this case, it must be removed from the PR.

**Review by lredor — CHANGES_REQUESTED** [review]
Waiting the use case data and steps to validate to review this PR

**Review by lredor — COMMENTED** [review]
Can you add this last line on all commits referencing issue [116] at the end of the commit message?
Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/116

File: plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java [hunk 1]
```diff
@@ -93,21 +115,17 @@ public DTableColumnLabelProvider(final DColumn column) {
...
-                }
+        
+        DTableElementStyle styleToApply = getBgStyleToApply(element);
+        if (styleToApply != null) {
+            final RGBValues rgb = styleToApply.getBackgroundColor();
+            if (rgb != null) {
+                result = VisualBindingManager.getDefault().getColorFromRGBValues(rgb);
             }
         }
+        // Background color is above selection color.
```
Comment (lredor) [inline plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java:126]: These 3 lines must be moved in previous commit `Use default selection color for table.`.

File: plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java [hunk 2]
```diff
@@ -93,21 +115,17 @@ public DTableColumnLabelProvider(final DColumn column) {
...
-                styleToApply = new DCellQuery(optionalCell.get()).getBackgroundStyleToApply();
-            } else if (element instanceof DLine) {
-                styleToApply = TableHelper.getBackgroundStyleToApply((DLine) element, column);
-            }
-            if (styleToApply != null && styleToApply.some()) {
-                final RGBValues rgb = styleToApply.get().getBackgroundColor();
-                if (rgb != null) {
-                    result = VisualBindingManager.getDefault().getColorFromRGBValues(rgb);
-                }
+
```
Comment (lredor) [inline plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java:118]: These lines must be moved in previous commit `Use default selection color for table.`.

**Comment by lredor** [comment]
The commit "[116] Align font type and font size in table between cell and axis." mixes some cleanup code" and real changes concerning "font type and font size". For the next time, it could be better to split that in two commits (to facilitate the review and to facilitate potential regressions analysis later).

**Review by lredor — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java (hunk 1 above)
Comment (mypsycho) [reply plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java:126]: Done.

File: plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/DTableViewerManager.java [hunk 3]
```diff
@@ -250,11 +248,17 @@ protected void createTreeViewer(final Composite theComposite) {
         treeViewer.getTree().setLinesVisible(true);
         treeViewer.getTree().setHeaderVisible(true);
 
+
         // Create a new CellFocusManager
-        final TreeViewerFocusCellManager focusCellManager = new TreeViewerFocusCellManager(treeViewer, new FocusCellOwnerDrawHighlighter(treeViewer));
+        final TreeViewerFocusCellManager focusCellManager = new TreeViewerFocusCellManager(treeViewer,
```
[the remark this answers is not in the document]
Comment (mypsycho) [reply plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/DTableViewerManager.java:253]: Modification reverted.

File: plugins/org.eclipse.sirius.doc/doc/Release_Notes.textile [hunk 4]
```diff
@@ -9,11 +9,28 @@ h2(#sirius7.3.0). Changes in Sirius 7.3.0
 h3. User-Visible Changes
 
 * <span class="label label-success">Added</span> A new preference has been added in the Sirius Diagram preference page. This preference allows the user to choose the arrangement of elements created on a diagram when they are not directly positioned by the user. In the preference page, users can choose between a diagonal, vertical or horizontal arrangement.
+* <span class="label label-info">Modified</span> In Sirius table, font type and default size are now based on Preferences Settings of **General > Appearance > Colors and Fonts**. Previously, cells font were always 'Arial' and of size '9' but column and line headers were based on Preferences Settings.
+* <span class="label label-info">Modified</span> In Sirius table, default colors has been changed to use Preferences Settings and selection coloring behaves like other Views and Editors. Previously, line headers were forced to grey, selected cell was using Menu color instead of Theme color and no color was indicating the row on mouse hover.
```
[the remark this answers is not in the document]
Comment (mypsycho) [reply plugins/org.eclipse.sirius.doc/doc/Release_Notes.textile:13]: That's a [gray](https://en.wiktionary.org/wiki/gray#Alternative_forms) question. Changed anyway.

[the remark this answers is not in the document]
Comment (lredor) [reply plugins/org.eclipse.sirius.doc/doc/Release_Notes.textile]: OK, you consider the change, but you have not regenerate the documentation. The "Html" file is wrong.

File: plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java [hunk 5]
```diff
@@ -220,15 +241,15 @@ public String getText(final Object element) {
      *            the DLine or DCell in which to look.
      * @return an optional cell
      */
-    protected Option<DCell> getDCell(final Object element) {
-        Option<DCell> optionalCell = Options.newNone();
+    protected DCell getDCell(final Object element) {
```
[the remark this answers is not in the document]
Comment (lredor) [reply plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java:258]: Is there a reason to not consider this remark in your last push ?

[the remark this answers is not in the document]
Comment (lredor) [reply plugins/org.eclipse.sirius.table.ui/src/org/eclipse/sirius/table/ui/tools/internal/editor/provider/DTableColumnLabelProvider.java]: OK, it's an internal class. So it's not really a blocker remark.

**Comment by lredor** [comment]
PR to rebase manually before I can merge.

**Comment by lredor** [comment]
This PR has been indirectly merged through PR #169

**Pull request closed without merging**